packages/matter.js/src/behavior/definitions/general-diagnostics/GeneralDiagnosticsServer.ts

```diff
@@ -39,10 +73,306 @@ export class GeneralDiagnosticsServer extends GeneralDiagnosticsBehavior {
         );
     }
 
+    /**
+     * Register a hardware fault. This convenience method updates the activeHardwareFaults attribute and sends out the
+     * HardwareFaultChanged event. Make sure to clear the fault when it is resolved.
+     * This method requires that the activeHardwareFaults attribute is activated when the cluster gets initialized.
+     *
+     * @param faultType The hardware fault to register.
+     */
+    registerHardwareFault(faultType: GeneralDiagnostics.HardwareFault) {
+        if (this.state.activeHardwareFaults === undefined) {
```

Comment:
This test happens 6x in this file alone.  Maybe add an "assertAttributeEnabled" or "requireAttribute" utility method to ClusterBehavior that returns the attribute value or throws?  The message could be "To use this feature, please enable attribute ${attributeName} by setting the value during initialization"